bot/utils/regex.py

```diff
@@ -10,3 +10,5 @@
     r"([a-zA-Z0-9\-]+)",                              # the invite code itself
     flags=re.IGNORECASE
 )
+
+END_PUNCTUATION_RE = re.compile("^.+?[.?!]$")
```

Comment:
I suggest avoiding a regex for this. Use `str.endswith((".", "?", "!"))`; it's simpler and therefore easier to understand.